tensorflow-framework/src/main/java/org/tensorflow/framework/utils/SparseTensor.java

```diff
@@ -0,0 +1,75 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.framework.utils;
+
+import org.tensorflow.Operand;
+import org.tensorflow.types.TInt64;
+import org.tensorflow.types.family.TType;
+
+/**
+ * This is a helper class that represents a sparse tensor who's attributes may be passed to
+ * {@link org.tensorflow.op.Ops#sparse} methods.
+ *
+ * @param <T> the type of the SparseTensor
+ */
+public class SparseTensor<T extends TType> {
```

Comment:
Shouldn't this live in `tensorflow-core-api`?

Reply:
Yes, 

Right now, it is a Java based class and not a true `Tensor` as in TF. I use it to hold the arguments `(indices,values,denseShape) ` that are fed to some of the `tf.sparse` operations, such as `tf.sparse.sparseTensorDenseAdd`. 

It is not clear to me yet, what the value of making it a true Tensor is, as all our `sparse` functions take the constitute attributes separately, instead of accepting the `SparseTensor` directly.

The concept of a `SparseTensor` in TF Python looks like a Pythonism to me, within the higher level Python code for the `sparse` operations, they break out the constituent parts before calling the low level `sparse` operation. The low level sparse functions never pass the `SparseTensor` as a whole.

Reply:
I think having a `SparseTensor` that mirrors the way python does it is fine (which this is, bar the sugar on the methods accepting it rather than it's constituent parts). We should document it as being not an actual tensor and shove it in core. Later we can modify the codegen to add overloads that accept it in `tf.sparse` rather than the components.

Reply:
`package org.tensorflow;` under `tensorflow-core-api`?

Reply:
I think so, with a documentation update to `Tensor` to note that it's a dense tensor. But I can see how it would be confusing to users, which is why in python it's in `tf.sparse`. I'm not sure of the utility of creating a separate package for a single class though. @karllessard what do you think?

Reply:
@Craigacp @karllessard, Have we come to consensus on how to handle SparseTensor?